robot-server/robot_server/instruments/instrument_models.py

```diff
@@ -0,0 +1,105 @@
+"""Request and response models for /instruments endpoints."""
+from __future__ import annotations
+
+from typing_extensions import Literal
+from typing import Optional, TypeVar, Union, Generic
+from enum import Enum
+from pydantic import BaseModel, Field
+from pydantic.generics import GenericModel
+
+from opentrons.hardware_control.instruments.ot3.instrument_calibration import (
+    GripperCalibrationOffset,
+)
+from opentrons.hardware_control.types import GripperJawState
+from opentrons.types import Mount
+
+from opentrons_shared_data.pipette.dev_types import (
+    PipetteName,
+    PipetteModel,
+    ChannelCount,
+)
+from opentrons_shared_data.gripper.dev_types import (
+    GripperName,
+    GripperModel,
+)
+
+InstrumentNameT = TypeVar("InstrumentNameT", bound=Union[GripperName, PipetteName])
+InstrumentModelT = TypeVar("InstrumentModelT", bound=Union[GripperModel, PipetteModel])
+InstrumentDataT = TypeVar("InstrumentDataT", bound=BaseModel)
+
+InstrumentType = Literal["pipette", "gripper"]
+
+
+# TODO (spp, 2023-01-03): use MountType from opentrons.types once it has extension type
+class MountType(str, Enum):
```

Comment:
In general, [I think we should avoid subclassing `Enum`s from `str`](https://opentrons.slack.com/archives/CTLS4MBB4/p1667947955558669).

In this case, it looks like we can just change the declaration without having to modify any other code.

```suggestion
class MountType(Enum):
```

Reply:
Interesting thread! 
But this is just a temporary enum until we add extension mount to `opentrons.types.MountType`. Still, the change is small and it might help in updating the main `MountType` so I'm going to make this change here. Thanks!